controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -451,38 +572,173 @@ func (pp *portPublisher) updateEndpoints(endpoints *corev1.Endpoints) {
 			}
 		}
 	}
+	pp.addresses = newAddressSet
 	pp.exists = true
+	pp.metrics.incUpdates()
+	pp.metrics.setPods(len(pp.addresses.Addresses))
+	pp.metrics.setExists(true)
+}
+
+func (pp *portPublisher) addEndpointSlice(slice *discovery.EndpointSlice) {
+	newAddressSet := pp.endpointSliceToAddresses(slice)
+	for id, addr := range pp.addresses.Addresses {
+		newAddressSet.Addresses[id] = addr
+	}
+
+	addAddress := make(map[ID]Address, len(newAddressSet.Addresses))
+	for id, newAddress := range newAddressSet.Addresses {
+		if _, ok := pp.addresses.Addresses[id]; !ok {
+			addAddress[id] = newAddress
+		}
+	}
+
+	if len(addAddress) > 0 {
+		add := AddressSet{
+			Addresses: addAddress,
+			Labels:    newAddressSet.Labels,
+		}
+		for _, listener := range pp.listeners {
+			listener.Add(add)
+		}
+	}
+
 	pp.addresses = newAddressSet
+	pp.exists = true
+	pp.metrics.incUpdates()
+	pp.metrics.setPods(len(pp.addresses.Addresses))
+	pp.metrics.setExists(true)
+}
+
+func (pp *portPublisher) updateEndpointSlice(oldSlice *discovery.EndpointSlice, newSlice *discovery.EndpointSlice) {
+	newAddressSet, oldAddressSet := pp.endpointSliceToAddresses(newSlice), pp.endpointSliceToAddresses(oldSlice)
+	if len(newAddressSet.Addresses) == 0 && len(oldAddressSet.Addresses) == 0 {
```

Comment:
This is a very special case that I've come across when I was playing around. When we don't have any endpoints left and we update a slice with nil ports and nil endpoints with a slice that has nil ports and nil endpoints, the dest-service spams `NoEndpoints`. This checks avoids that.

Reply:
With my suggestion below to not use NoEndpoints, this won't be an issue.